keps/sig-scheduling/5471-enable-sla-based-scheduling/README.md

```diff
@@ -0,0 +1,1125 @@
+---
+# Inference service requires SLA >= 950 with 30s grace period
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: inference-service
+spec:
+  template:
+    spec:
+      tolerations:
+      - key: node.kubernetes.io/sla
+        operator: Ge
+        value: "950"
+        effect: NoExecute
+        tolerationSeconds: 30
+```
+
+#### Story 3 — AI training workload balancing cost and reliability
+
+As an ML engineer running large distributed training, I want to run most worker pods on cheaper spot GPU nodes, but keep certain roles (e.g., parameter servers, checkpoint writers) on SLA ≥ 99.9% on-demand GPUs.
+
+With numeric tolerations, I can opt-in only the pods that are safe to run on spot, while letting the cluster's default taints repel all others.
+
+**Example Configuration:**
+
+```yaml
+# Parameter server requires ultra-high reliability
+apiVersion: v1
+kind: Pod
+metadata:
+  name: parameter-server
+spec:
+  tolerations:
+  - key: node.kubernetes.io/sla
+    operator: Ge
+    value: "999"  # 99.9% SLA
+    effect: NoSchedule
+  containers:
+  - name: ps
+    resources:
+      requests:
+        nvidia.com/gpu: 1
+---
+# Training workers can tolerate spot nodes
+apiVersion: v1
+kind: Pod
+metadata:
+  name: training-worker
+spec:
+  tolerations:
+  - key: node.kubernetes.io/sla
+    operator: Ge
+    value: "800"  # 80% SLA acceptable
+    effect: NoSchedule
+  containers:
+  - name: worker
+    resources:
+      requests:
+        nvidia.com/gpu: 4
+```
+
+#### Story 4 — DRA GPU claim management
+
+As a DRA driver implementer, I want to combine device resource claims with node SLA constraints so that GPU claims can only bind to nodes meeting a minimum reliability, unless the workload explicitly tolerates lower values.
+
+This ensures DRA allocations are both resource-correct and reliability-compliant.
+
+**Example Configuration:**
+
+```yaml
+# DRA claim with SLA constraints
+apiVersion: resource.k8s.io/v1alpha4
+kind: ResourceClaim
+metadata:
+  name: gpu-claim-high-sla
+spec:
+  devices:
+    requests:
+    - name: gpu
+      deviceClassName: nvidia-a100
+---
+# Pod using DRA claim with SLA requirements
+apiVersion: v1
+kind: Pod
+metadata:
+  name: dra-workload
+spec:
+  resourceClaims:
+  - name: gpu-claim
+    resourceClaimName: gpu-claim-high-sla
+  tolerations:
+  - key: node.kubernetes.io/sla
+    operator: Ge
+    value: "950"  # Ensure GPU nodes meet SLA requirements
+    effect: NoSchedule
+  containers:
+  - name: ml-workload
+    resources:
+      claims:
+      - name: gpu-claim
+```
+
+### Notes/Constraints/Caveats (Optional)
+
+- **Integer-Only Support**: The implementation supports signed 64-bit integers only. Decimal values (e.g., `"95.5"`) will be rejected by API validation when using numeric operators.
+
+- **Parsing Requirements**: Both taint value and toleration value must be parseable as integers for numeric operators (`Lt`, `Le`, `Ge`, `Gt`). If either fails parsing, the toleration does not match.
```

Comment:
Again, we can't validate the taint, right? This is a valid taint:

`foo=95.5:NoSchedule`

So I think we have to emphasize that parsing validation will be enforced on the toleration (*when* it declares one of the new, proposed operators)

Reply:
Correct, We cannot. A taint like `foo=95.5:NoSchedule` is valid because taint values follow label value syntax, which allows. The numeric parsing/validation is enforced on tolerations only. I'll make sure to make this clear in the KEP.

Reply:
Addressed.